plugins/debuginfo-install.py

```diff
@@ -34,9 +36,21 @@ class DebuginfoInstall(dnf.Plugin):
     def __init__(self, base, cli):
         """Initialize the plugin instance."""
         super(DebuginfoInstall, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
         if cli is not None:
             cli.register_command(DebuginfoInstallCommand)
 
+    def config(self):
+        cp = self.read_config(self.base.conf, PLUGIN_CONF)
+        autoupdate = cp.getboolean('main','autoupdate')
+
+        if autoupdate:
+            # allow update of already installed debuginfo packages
+            dbginfo = [p.name for p in dnf.sack.rpmdb_sack(self.base).query()
```

Comment:
>    What's the reason of creating a new sack? base.sack is not loaded, yet, right?

Two reasons - base.sack isn't available at the moment, it's loaded after config and iterating over base.sack.query().installed() is slower than over rpmdb_sack(self.base).query() although they both contain the same set of packages.

>    Why the filtering of the name happens afterwards not in the query itself? Not query().filter(name__glob="*-debuginfo")

I for sure tried this but for a purpose ended up with filtering list; most likely it was slower.
Checking now I can't see a reason for this (maybe something has been fixed in the meantime? :) so changing it to query filter.